File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0
    dp = [[0] * i for i in range(1, len(triangle) + 1)]
    dp[0] = triangle[0]
    
    for i in range(1, len(triangle)):
        for j in range(len(triangle[i])):
            if j == 0:
                dp[i][j] = triangle[i][j] + dp[i - 1][j]
            elif j == len(triangle[i]) - 1:
                dp[i][j] = triangle[i][j] + dp[i - 1][j - 1]
            else:
                dp[i][j] = triangle[i][j] + max(dp[i - 1][j], dp[i - 1][j - 1])
    
    answer = max(dp[-1])  
    return answer

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.js
function solution(triangle) {
    let answer = 0;
    let Arr =[];
    let Sum=[];
    for(var i=0; i<triangle.length; i++){
        Sum[i]=[];
        Arr[i]=[];
        for(var j=0; j<triangle[i].length; j++)  Arr[i].push(parseInt(triangle[i][j]))
    }
    Sum[triangle.length-1]=Arr[triangle.length-1];
    Sum[triangle.length-1]=Arr[triangle.length-1];
    for(var i= 1; i<triangle.length; i++){
        for(var j=0; j<Arr[triangle.length-i-1].length;j++){
            Sum[triangle.length-i-1][j]=Math.max(Sum[triangle.length-i][j],Sum[triangle.length-i][j+1])+Arr[triangle.length-i-1][j]
        }
    }
    answer=Sum[0][0];
    return answer;
}